### PR TITLE
update to u-root 0.12.0 and more cleanups

### PR DESCRIPTION
**Pull request by majst01**
TODO:

depends on: https://github.com/metal-stack/go-hal/pull/60

- [x] use go-1.21
- [x] switch to slog
- [x] test
  - [x] machine creation worked

### CODE REVIEW

**Comment by majst01** [comment]
```
[   12.422948] cgroup: Unknown subsys name 'net_cls'
[   12.427728] cgroup: Unknown subsys name 'hugetlb'
/bbin/uinit: errdefaultsh-5.1# uinit
uinit: error while loading shared libraries: libpcap.so.0.8: cannot open shared object file: No such file or directory
defaultsh-5.1# 
```

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+
+RUN apk add \
+	binutils \
+	coreutils \
+	curl \
+	gcc \
+	git \
+	make \
+	musl-dev \
+    libpcap-dev
```
Comment (Gerrit91) [inline Dockerfile:11]: Missing indent

File: Makefile [hunk 2]
```diff
@@ -1,18 +1,45 @@
+.ONESHELL:
```
Comment (Gerrit91) [inline Makefile:1]: Really required?

File: Dockerfile (hunk 1 above)
Comment (majst01) [reply Dockerfile]: done

File: Makefile (hunk 2 above)
Comment (majst01) [reply Makefile:1]: don't want to change this now

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**